### DISCUSSION TITLE
Restart the server automatically when you modify next.config.js

### DISCUSSION THREAD

**Discussion by kud** [vercel/next.js#19030] [Ideas]
It could be great to restart itself the server when you modify the `next.config.js` file.

For the moment I'm doing this: `nodemon --watch next.config.js` which starts `next dev` and watches the file `next.config.js`.

**Comment by shixin-guo** [comment]
I also love this ideo

**Comment by balazsorban44** [comment]
This has been added in version 13.3: https://nextjs.org/blog/next-13-3#other-improvements:~:text=Fast%20Refresh%20for%20next.config.js
